src/main/scala/wdl4s/types/WdlArrayType.scala

```diff
@@ -15,26 +15,33 @@ sealed trait WdlArrayType extends WdlType {
   private def coerceIterable(values: Seq[Any]): WdlArray = values match {
     case s:Seq[Any] if s.nonEmpty =>
       val coerced = s.map {memberType.coerceRawValue(_).get}
-      WdlArray(WdlArrayType(coerced.head.wdlType), coerced)
-    case _ => WdlArray(WdlArrayType(memberType), Seq())
+      WdlArray(this, coerced)
+    case _ => WdlArray(this, Seq())
   }
 
-  protected def coercionMaker(allowEmpty: Boolean): PartialFunction[Any, WdlValue] = {
+  private val allowEmpty = !guaranteedNonEmpty
```

Comment:
Not really keen on this whole `guaranteedNonEmpty` thing but it seems to be a preexisting condition and those aren't covered anymore so consider this a TOL